ethsigner/src/main/java/tech/pegasys/ethsigner/requesthandler/JsonRpcRequestHandler.java

```diff
@@ -12,12 +12,42 @@
  */
 package tech.pegasys.ethsigner.requesthandler;
 
+import tech.pegasys.ethsigner.http.HttpResponseFactory;
 import tech.pegasys.ethsigner.jsonrpc.JsonRpcRequest;
+import tech.pegasys.ethsigner.jsonrpc.response.JsonRpcError;
+import tech.pegasys.ethsigner.jsonrpc.response.JsonRpcErrorResponse;
 
+import io.netty.handler.codec.http.HttpResponseStatus;
 import io.vertx.core.http.HttpServerRequest;
+import io.vertx.core.json.Json;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
-@FunctionalInterface
-public interface JsonRpcRequestHandler {
+public abstract class JsonRpcRequestHandler {
 
-  void handle(HttpServerRequest httpServerRequest, JsonRpcRequest rpcRequest);
+  public abstract void handle(HttpServerRequest httpServerRequest, JsonRpcRequest rpcRequest);
+
+  private static final Logger LOG = LoggerFactory.getLogger(JsonRpcRequestHandler.class);
+
+  protected final HttpResponseFactory responder;
+
+  public JsonRpcRequestHandler(HttpResponseFactory responder) {
+    this.responder = responder;
+  }
+
+  protected void reportError(
```

Comment:
Why did you replace JsonRpcErrorReporter with this method?

Reply:
So we could have kept JsonRpcErrorReporter - but given it worked with "JsonRpcRequest" (which isn't available in the "TransactionTransmitter" - seemed better to make this a local capability.